agent/dns_test.go

```diff
@@ -3832,6 +3832,45 @@ func TestDNS_Recurse_Truncation(t *testing.T) {
 	}
 }
 
+func TestDNS_Recursor_StrategyRandom(t *testing.T) {
```

Comment:
Since the code this is testing is in `agent/dns/dns.go` , I guess these two tests should move to `agent/dns/dns_test.go`